### PR TITLE
Fix make lint

### PR DESCRIPTION
**Pull request by amwat**
lint inside go_container.

/cc @BenTheElder

### CODE REVIEW

**Comment by amwat** [comment]
also IIUC plumbing GOOS through actually breaks on mac, since our container is running inside a linux vm.
but I might be missing something @BenTheElder ?

File: hack/go_container.sh [hunk 1]
```diff
@@ -38,22 +38,7 @@ DOCKER="${DOCKER:-"$(which docker || which podman || echo "docker")"}"
 # ========================== END SCRIPT SETTINGS ===============================
 
 # autodetects host GOOS and GOARCH and exports them if not set
-detect_and_set_goos_goarch() {
```
Comment (BenTheElder) [inline hack/go_container.sh]: No. This breaks make install

File: hack/verify/lint.sh [hunk 2]
```diff
@@ -36,7 +34,8 @@ LINTS=(
 )
 ENABLE=$(sed 's/ /,/g' <<< "${LINTS[@]}")
 # first for the repo in general
-GO111MODULE=on bin/golangci-lint --disable-all --enable="${ENABLE}" run ./pkg/... ./cmd/... .
+SOURCE_DIR="${REPO_ROOT}" hack/go_container.sh \
+  /out/golangci-lint --disable-all --enable="${ENABLE}" run ./pkg/... ./cmd/... .
 # ... and then for kindnetd, which is only on linux
 SOURCE_DIR="${REPO_ROOT}/images/kindnetd" GOOS="linux" "${REPO_ROOT}/hack/go_container.sh" \
```
Comment (BenTheElder) [inline hack/verify/lint.sh]: We set GOOS here

File: hack/verify/lint.sh [hunk 3]
```diff
@@ -23,8 +23,6 @@ cd "${REPO_ROOT}"
 # build golangci-lint
 SOURCE_DIR="${REPO_ROOT}/hack/tools" hack/go_container.sh \
   go build -o /out/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
-SOURCE_DIR="${REPO_ROOT}/hack/tools" GOOS="linux" hack/go_container.sh \
```
Comment (BenTheElder) [inline hack/verify/lint.sh]: And here

**Comment by BenTheElder** [comment]
It could make sense to autodetect GOOS only during make build, but the current change leaves that broken.

**Pull request merged**